### PR TITLE
Refactor visualizer

### PR DESCRIPTION
**Pull request by denix56**
Improve performance, fix compilation issues

Finally, I have found time to split my last pull request in several parts.

### CODE REVIEW

**Comment by taketwo** [comment]
I'm afraid that besides from your refactorings, the commit undoes several of your recently merged pull requests (#2038, #2077).

**Comment by taketwo** [comment]
Thanks. Could you also clarify these guards `#if VTK_RENDERING_BACKEND_OPENGL_VERSION < 2` that you added. Why? Was there a bug? Also pointers to VTK documentation.

**Comment by denix56** [comment]
I added this guard, because in one of the recent VTK commits (somewhere in September/October) the function was removed (it was not defined in OpenGL2 at all, it had empty body)

File: visualization/src/pcl_visualizer.cpp [hunk 1]
```diff
@@ -3056,9 +3060,10 @@ pcl::visualization::PCLVisualizer::addPolygonMesh (const pcl::PolygonMesh &poly_
   pcl::fromPCLPointCloud2 (poly_mesh.cloud, *point_cloud);
   poly_points->SetNumberOfPoints (point_cloud->points.size ());
 
-  size_t i;
-  for (i = 0; i < point_cloud->points.size (); ++i)
-    poly_points->InsertPoint (i, point_cloud->points[i].x, point_cloud->points[i].y, point_cloud->points[i].z);
+  for (size_t i = 0; i < point_cloud->points.size (); ++i) {
```
Comment (taketwo) [inline visualization/src/pcl_visualizer.cpp]: New line

File: visualization/include/pcl/visualization/pcl_visualizer.h [hunk 2]
```diff
@@ -293,7 +294,7 @@ namespace pcl
         /** \brief Disables the Orientatation Marker Widget so it is removed from the renderer */
         void
         removeOrientationMarkerWidgetAxes ();
-
+
```
Comment (SergioRAgostinho) [inline visualization/include/pcl/visualization/pcl_visualizer.h]: Did we add some trailing spaces here?

File: visualization/src/pcl_visualizer.cpp [hunk 3]
```diff
@@ -4079,7 +4085,7 @@ pcl::visualization::PCLVisualizer::renderViewTesselatedSphere (
 
     poses.push_back (pose_view);
 
-  }
+}
 }
```
Comment (SergioRAgostinho) [inline visualization/src/pcl_visualizer.cpp]: Indentation here seemed correct. Having two `}` in consecutive lines with no indentation doesn't. What happened exactly?

File: visualization/src/pcl_visualizer.cpp [hunk 4]
```diff
@@ -3979,7 +3985,7 @@ pcl::visualization::PCLVisualizer::renderViewTesselatedSphere (
       triangle->GetPoints ()->GetPoint (1, p1);
       triangle->GetPoints ()->GetPoint (2, p2);
       visible_area += vtkTriangle::TriangleArea (p0, p1, p2);
-    }
+  }
```
Comment (SergioRAgostinho) [inline visualization/src/pcl_visualizer.cpp]: This also looks like a mistake. The indentation was similar to the opening bracket, before the change.

**Comment by SergioRAgostinho** [comment]
This is not exactly a refactoring, but more like a code style cleanup. I'll probably change the commit message to that if it's ok.

**Comment by SergioRAgostinho** [comment]
Changing the status back to "waiting feedback" because there's still two comments of my review left to be addressed.

File: visualization/src/pcl_visualizer.cpp (hunk 1 above)
Comment (amnanajib) [reply visualization/src/pcl_visualizer.cpp]: Hi, 
i am using pcl 1.8.0. Should i only then change the pcl_visualizer.cpp to fix the probblem? And if yes:
my cpp file has a different structure, meaning the following:
```

  pcl::fromPCLPointCloud2 (poly_mesh.cloud, *point_cloud);
  poly_points->SetNumberOfPoints (point_cloud->points.size ());

  size_t i;
  for (i = 0; i < point_cloud->points.size (); ++i)
    poly_points->InsertPoint (i, point_cloud->points[i].x, point_cloud->points[i].y, point_cloud->points[i].z);

```

Any help please?

**Pull request merged**